docs/harvester-network.md

```diff
@@ -0,0 +1,43 @@
+# Introduction of Harvester Network
+Harvester supports two kinds of networks by version 0.2, management network and VLAN. 
+
+## Principles
+### Management network
+
+Harvester  adopts [flannel](https://github.com/flannel-io/flannel)  as the default CNI to implement management network. It's an internal network, exactly, that means it should be within cluster if you want to access a VM via the management network.
+
+### VLAN
+
+[Harvester network controller](https://github.com/rancher/harvester-network-controller) provides an VLAN network implementation for Harvester with the help of [multus](https://github.com/k8snetworkplumbingwg/multus-cni) and [bridge CNI](https://www.cni.dev/plugins/current/main/bridge/).
+
+Here is the VLAN network principle diagram. 
+
+<img src="https://raw.githubusercontent.com/yaocw2020/img/master/image-20210105185112481.png" alt="image-20210105185112481" style="zoom: 50%;" />
+
+> Note that the external switch ports have to be set as trunk or hybrid type and permit the specified VLANs.
+
+## How to Use
+
+- Go to **Advanced > Networks** tab, jump to **Setting > vlan** to enable VLAN.
+
+  ![image-20210408233828285](https://raw.githubusercontent.com/yaocw2020/img/master/image-20210408233828285.png)
+
+  ![image-20210408233858097](https://raw.githubusercontent.com/yaocw2020/img/master/image-20210408233858097.png)
+
+- Go to **Advanced > Networks**, if there are some errors shown above, it's required to fix the errors. There are two methods. 
+
+  ![image-20210408235847675](https://raw.githubusercontent.com/yaocw2020/img/master/image-20210408235847675.png)
+
+  The first method, go to **Setting > vlan** to disable vlan, then enable vlan with an valid default physical NIC for all hosts.
+
+  The second method, jump to **Host** according to the error hints, go to **Network** tab to select an available physical NIC for the host. 
+
+  ![image-20210408235822223](https://raw.githubusercontent.com/yaocw2020/img/master/image-20210408235822223.png)
+
+  If there is no error, click **Create** Button to create a VLAN network.
+
+  ![image-20210408232931111](https://raw.githubusercontent.com/yaocw2020/img/master/image-20210408232931111-20210409000151223.png)
+
+- When creating or editing VM, select the **Networks** tab, you can change the existing network or add new network with the VLAN network created before. 
+
+  ![image-20210408233240281](https://raw.githubusercontent.com/yaocw2020/img/master/image-20210408233240281.png)
```

Comment:
- Enable the default VLAN network configuration via ...
- Input a valid default physical NIC name of the VLAN network, the first physical NIC name of each Harvester node always defaults to `eth0`, it is recommended to choose a separate NIC for the VLAN other than the one used for the management network(the one selected during the Harvester installation) for better network performance and isolation. (Note: modify the default VLAN network setting will not change the existing configured host networks)
- optional) users can always customize each node's VLAN network configuration via ...
- create a new VLAN network via ... (Note: You can use VLAN 1 for simple installations where there is no VLAN trunking, but good practice is to not put user data on VLAN 1 in a multi-VLAN environment)
- Creating a VM and add the network configurations
    - only the first network card will be enabled by default, the user can either choose to use a management network or a VLAN network. (Note: will need to select `Install guest agent` option in the `Advanced Options` tab to get the VLAN network IP address from the Harvester UI)
 ![image](https://user-images.githubusercontent.com/4569037/114151440-b13d9000-994f-11eb-88c3-7cda38174a02.png)
    - users can choose to add one or multiple network cards, the additional network card configurations can be set via cloud-init network data. e.g.
```YAML
version: 1
config:
  - type: physical
    name: enp1s0 # name is various upon OS image
    subnets:
      - type: dhcp
  - type: physical
    name: enp2s0 
    subnets:
      - type: DHCP
```